### PR TITLE
add option to specificy max size of stack traces

### PR DESCRIPTION
**Pull request by kjarmicki**
A really simple implementation that just refuses any new entries above a certain limit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,15 +41,35 @@ const { stop } = blocked(fn, options)
 |---|---|---|
 |`trimFalsePositives`|*falsy*| eliminate a class of false positives (experimental) |
 |`threshold`| *20* | minimum miliseconds of blockage to report. supported for parity with [`blocked`](https://www.npmjs.com/package/blocked)|
+|`maxResourcesSize`| *undefined* | maximum amount of stack traces with resource details kept in memory. see the next section for details |
```
Comment (naugtur) [inline README.md]: The name could be clearer.

`keepResources` or `resourcesCap`

File: README.md [hunk 2]
```diff
@@ -41,15 +41,35 @@ const { stop } = blocked(fn, options)
...
+  if (type === 'HTTPPARSER' && resource) {
+    // resource structure assumes Node 10.x
+    console.log(`URL related to blocking operation: ${resource.resource.incoming.url}`)
+  }
+}, {maxResourcesSize: 100})
+```
+
+Note that resource structure is a subject to change and may vary between Node versions.
+
+ After you've identified a problematic URL, you can wrap your handlers' code in `setImmediate` which should make the stack point to something meaningful.
```
Comment (naugtur) [inline README.md]: I'd like to keep the set* suggestion as the default thing to try. It's more likely to work if you already have a suspect.
I'd move it back up and describe resources usecase as a separate paragraph.

File: test/resources.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+'use strict'
+const blocked = require('..')
+const http = require('./cases/http')
+
+ blocked((time, stack, {type, resource}) => {
```
Comment (naugtur) [inline test/resources.js]: Some assertions would be nice, to at least make it fail with a descriptive message when I remove the feature.

**Comment by naugtur** [comment]
I reviewed and I think the approach is good.
Did you check if it addresses your usecase for sure?

I wonder if storing the first N of resources (which is what effectvely happens if destroy isn't called) is the best choice instead of storing N most recent resources. 
But it seems better - less wasted memory, more immediate results.

You can try iterating on the documentation if you want, otherwise I'll address my comments some time soon and release.

Thanks!

**Comment by kjarmicki** [comment]
> I wonder if storing the first N of resources (which is what effectvely happens if destroy isn't called) is the best choice instead of storing N most recent resources.
But it seems better - less wasted memory, more immediate results.

That's true, but it might not be what we want here.
It's implicit that resources that are already stored are the ones that will take the most time, so it's possible that within them there are the most blocking ones. Keeping only the most recent ones around allows for a situation where a lot of new fast async actions wipe out longer running, potentially blocking ones from storage.

I'll adjust documentation and test according to your reviews in a few days. Thanks for support :)

**Comment by naugtur** [comment]
Thank you for contributing!
I'll use it in debugging-aid to improve output there.

**Pull request merged**